### ISSUE TITLE
[static check] super() not allowed in a non-constructor method

### ISSUE THREAD

**Issue by shicks** [google/closure-compiler#1708]
The following code should produce an error but [does not](https://closure-compiler.appspot.com/home#code%3D%252F%252F%2520%253D%253DClosureCompiler%253D%253D%250A%252F%252F%2520%2540compilation_level%2520WHITESPACE_ONLY%250A%252F%252F%2520%2540output_file_name%2520default.js%250A%252F%252F%2520%253D%253D%252FClosureCompiler%253D%253D%250A%250A%252F%252F%2520ADD%2520YOUR%2520CODE%2520HERE%250Aclass%2520A%2520%257B%250A%2520%2520init()%2520%257B%257D%250A%257D%250Aclass%2520B%2520extends%2520A%2520%257B%250A%2520%2520init()%2520%257B%2520super()%253B%2520%257D%250A%257D%250A):

```js
class A {
  init() {}
}
class B extends A {
  init() {
    super();
  }
}
```

When run in Chrome, this code fails with "JS ERROR: Uncaught SyntaxError: 'super' keyword unexpected here", but instead the transpiled output is

```js
var A = function() {
};
A.prototype.init = function() {
};
var B = function(var_args) {
  A.apply(this, arguments);
};
$jscomp.inherits(B, A);
B.prototype.init = function() {
  A.prototype.init.call(this);
};
```

**Comment by Dominator008** [comment]
IIRC the specs changed and this became illegal after we implemented the transpilation. I noticed the problem but probably didn't fix it.

**Comment by ChadKillingsworth** [comment]
This should be a pretty easy one.

**Comment by Dominator008** [comment]
Closed by https://github.com/google/closure-compiler/commit/2000025c9c0b1bc209f8365a8590ca34e61218f0.
